cli/test/lib/tasks/dependency_spec.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * as of Webpack 5, dependencies that are polyfilled through the Provide plugin must be defined inside the CLI
+ * in order to guarantee there is a version of the dependency accessible by the cypress CLI, either in the cypress directory
+ * or the root of their project. Currently, these two dependencies are 'buffer' and 'process'
+ */
+describe('dependencies', () => {
+  it('process dependency exists in package.json and is available', () => {
+    const { dependencies } = require('../../../package.json')
+
+    expect(dependencies.process).to.be.ok
+
+    const process = require('process')
+
+    expect(typeof process).to.equal('object')
+  })
+
+  it('buffer dependency exists in package.json and is available', () => {
+    const { dependencies } = require('../../../package.json')
+
+    expect(dependencies.buffer).to.be.ok
+
+    const buffer = require('buffer')
```

Comment:
Should we assert that 

```js
require('node:buffer') !== require('buffer')
```

That would verify the `Buffer` you get is not the core Node.js one. You can use `require('node:<module>')` to get the core one.

Reply:
that might be a good idea, considering we really want to just make sure the `buffer` is installed in the package through the `package.json` and its accessible